ui/admin-portal/src/app/components/util/intake/IntakeComponentTabBase.ts

```diff
@@ -176,6 +182,116 @@ export abstract class IntakeComponentTabBase implements OnInit {
+    const dateMatch = notes.match(/Verification Date:\s*([0-9]{4}-[0-9]{2}-[0-9]{2})/);
+    return dateMatch ? new Date(dateMatch[1]) : null;
+  }
+
+  /**
+   * Determines the class and tooltip message for an exam score based on its value.
+   * The method categorizes the score into three ranges: below 60, between 60 and 89, and 90 or higher.
+   *
+   * @param {string} score - The score to be evaluated (string representation).
+   * @returns {Object} - An object containing the class name and tooltip message.
+   *    - 'text-mute' and 'Pending' for invalid or missing scores.
+   *    - 'text-danger' and 'Below requirement' for scores less than 60.
+   *    - 'text-warning' and 'Needs verification' for scores between 60 and 89.
+   *    - 'text-success' and 'Meets language requirements' for scores 90 or higher.
+   */
+  getExamInfo(score: string) {
+    let className = 'text-mute';
+    let tooltip = 'Pending. Score is not provided or invalid.';
+
+    if (score === null || score === undefined || isNaN(parseFloat(score))) {
+      // Handle null, undefined, or non-numeric scores
+      return { className, tooltip };
+    }
+
+    const numericScore = parseFloat(score);
+
+    if (numericScore < 60) {
+      className = 'text-danger';
+      tooltip = 'Below requirement. Score is < 60.';
+    } else if (numericScore >= 60 && numericScore < 90) {
+      className = 'text-warning';
+      tooltip = 'Needs verification against the language requirement. Score is between 60 and 89.';
+    } else {
+      className = 'text-success';
+      tooltip = 'Meets language requirements. Score is 90 or higher.';
+    }
+
+    return { className, tooltip };
+  }
+  /**
+   * Determines the label for an exam based on whether it is the most recent or has the highest score.
+   * The label identifies the exam as 'Best & Newest Score', 'Newest Score', 'Best Score', or 'DET Official'.
+   *
+   * @param {CandidateExam} exam - The exam object to evaluate.
+   * @returns {string} - The label for the exam.
+   *    - 'Best & Newest Score' for the exam with the highest score and the most recent date.
+   *    - 'Newest Score' for the most recent exam only.
+   *    - 'Best Score' for the exam with the highest score only.
+   *    - 'DET Official' for any other exam.
+   */
+  getExamLabel(exam: CandidateExam): string {
+    const mostRecent = this.getMostRecentDetOfficialExam(this.candidateIntakeData?.candidateExams);
+    const highestScore = this.getHighestScoreDetOfficialExam(this.candidateIntakeData?.candidateExams);
+
+    if (exam === mostRecent && exam === highestScore) {
+      this.examLabels[exam.id] = 'Best & Newest Score';
+    } else if (exam === mostRecent) {
+      this.examLabels[exam.id] = 'Newest Score';
+    } else if (exam === highestScore) {
+      this.examLabels[exam.id] = 'Best Score';
+    } else {
+      this.examLabels[exam.id] = 'DET Official';
+    }
+    return this.examLabels[exam.id];
+  }
```

Comment:
Great work @EhsanEhrari - I find this much easier to follow - and I think others will as well.